InnerEye/ML/Histopathology/utils/metrics_utils.py

```diff
@@ -58,33 +58,36 @@ def select_k_tiles(results: Dict, n_tiles: int = 5, n_slides: int = 5, label: in
             scores.append(results[attn_col][slide_idx][0][t_idx])
         # slide_ids are duplicated
         k_idx.append((results[slide_col][slide_idx][0],
-                      results[prob_col][slide_idx].item(),
+                      results[prob_col][slide_idx],
                       k_tiles, scores))
     return k_idx
 
 
-def plot_scores_hist(results: Dict, prob_col: str = ResultsKey.PROB,
+def plot_scores_hist(results: Dict, prob_col: str = ResultsKey.PROB_CLASS,
                      gt_col: str = ResultsKey.TRUE_LABEL) -> plt.figure:
     """
     :param results: List that contains slide_level dicts
     :param prob_col: column name that contains the scores
     :param gt_col: column name that contains the true label
     :return: matplotlib figure of the scores histogram by class
     """
-    pos_scores = [results[prob_col][i][0].cpu().item() for i, gt in enumerate(results[gt_col]) if gt == 1]
-    neg_scores = [results[prob_col][i][0].cpu().item() for i, gt in enumerate(results[gt_col]) if gt == 0]
-    fig, ax = plt.subplots()
-    ax.hist([pos_scores, neg_scores], label=['1', '0'], alpha=0.5)
+    n_classes = len(results[prob_col][0])
+    scores_class = []
+    for j in range(n_classes):
+        scores = [results[prob_col][i][j].cpu().item() for i, gt in enumerate(results[gt_col]) if gt == j]
+        scores_class.append(scores)
+    fig, ax = plt.subplots() 
+    ax.hist(scores_class, label=[str(i) for i in range(n_classes)], alpha=0.5)
     ax.set_xlabel("Predicted Score")
     ax.legend()
     return fig
 
 
-def plot_attention_tiles(slide: str, score: float, paths: List, attn: List, case: str, ncols: int = 5,
+def plot_attention_tiles(slide: str, score: List[float], paths: List, attn: List, case: str, ncols: int = 5,
```

Comment:
Very minor suggestion to rename `score` to `scores` for clarity that it's now meant to be a collection.

Reply:
Done